scripts/copper/slab/prepare.py

```diff
@@ -83,12 +89,47 @@ def _prepare_free_layer_convergence(atom: Atoms, data_path: Path) -> None:
     maybe_copy_files_to_hpc(data_path, PosixPath(data_path.as_posix()))
 
 
+def _prepare_slab_width_convergence(
+    atom: Atoms,
+    data_path: Path,
+) -> None:
+    calculators = list[Castep]()
+    prepare_clean_directory(data_path)
+
+    slab = get_surface(
+        atom,
+        (1, 1, 1),
+        n_fixed_layer=4,
+        n_free_layer=4,
+        n_vacuum_layer=5,
+    )
+
+    slab_position = get_top_position(slab)
+    rel_position = [0, 0, 1.0]
+    na_atom = prepare_na_atom(slab_position, rel_position)
+
+    for width in range(1, 9):
+        adsorbate = prepare_adsorbate(na_atom, slab, width)
+        config = CastepConfig(
+            data_path / f"width_{width}",
+            "slab_width",
+        )
+        params = SlabOptimizationParams(n_k_points=10, xc_functional="WC")
```

Comment:
n_k_points should also scale according to ceil(10 / width), since if we have a 2x2 slab n_k_points means a 20x20 supercell, but we only need a 10x10 supercell. Note I would use more than 10x10 just to be safe (12x12 is fine)